MetaMask/metamask-extension
# updated onClick of close button to call the same autohide fxn as the …

…NFT import removal

## Explanation
Referenced issue:
https://github.com/MetaMask/metamask-extension/issues/18485

Problem: after adding an NFT, I cannot close the Toast Notification by clicking on the cross sign.

There's an `onAutoHide` fxn that is used in the removeNftMessage `<ButtonIcon/>` component to close the toast notification that wasn't being called in the newNftAddedMessage component.  Calling this in an onClick allows the toast notification to close without relying on the autohide duration.

## Screenshots/Screencaps



### Before

![oldNFTImportButton](https://user-images.githubusercontent.com/33137497/230534187-e442ee74-20cd-4be3-88fd-19f337737e86.gif)

### After

![newNFTImportButton](https://user-images.githubusercontent.com/33137497/230533650-0e9ea151-4adc-45ca-ac05-d90a8a28283f.gif)

## Manual Testing Steps

1. Checkout and build `yarn start`
2. I used the E2E test app to deploy a collection and mint a test NFT on Goerli here https://metamask.github.io/test-dapp/ if you don't have an NFT
3. Once your connected wallet owns an NFT, input the collection + NFT ID into the fields after clicking import
4. Try to click the `x` to close the toast notification, should work on this branch but not `develop`.  The remove nft toast notification should work on both branches

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
